### ISSUE TITLE
Separate missing value options on summary dialogs

### ISSUE THREAD

**Issue by dannyparsons** [IDEMSInternational/R-Instat#6012]
I have a suggested change for missing value options on dialogs that produce summaries. Since videos will be made soon showing these dialogs I want to raise this now and if agreed implement quickly before videos are started.

Currently on e.g. the Column Summaries dialog there is a checkbox to exclude missing values, and then detailed missing value options a tab on the Summaries subdialog. There's several confusing things about this:

- If I don't check the missing check box then any missing options chosen on the sub dialog are ignored - but there's no way of knowing this from the dialog or sub dialog
- From the main dialog there's no indication that there are more missing options and its not very logical that I need to look on the Summaries sub dialog for this
- In general, there is a disconnect between the missing option on the main dialog and sub dialog

To resolve these I suggest a separate "Missing Options" sub dialog which contains the Missing tab currently on the Summaries sub dialog. The missing checkbox on the main dialog remains. The Missing Options button is placed next to the missing checkbox and is only enabled if the check box is checked. This then ensures missing options are only available when checked.
The functionality of all the controls stays exactly the same.

The only down side I see to this is that it uses a bit more space on the main dialog and makes it look a little bit more complex but I think this is worth it to have more clarity on the missing value options.

If it's agreed, @shadrackkibet might be able to do this quite quickly as he worked on the missing options originally? There are a number of affected dialogs but the work is essentially the same on all of them.

**Comment by shadrackkibet** [comment]
@rdstern what are your views on this? I am happy to proceed with this task.

**Comment by rdstern** [comment]
I was very pleased with this suggested changes.  I have felt that the presence of the Missing checkbox on the main dialogue, separated from the missing tab, on the sub-dialogue, was a bit confusing.  And the issue is becoming increasingly important, because of the work needed on at least documenting the climate normals.

**Comment by rdstern** [comment]
As @dannyparsons says, there are a number of dialogues with this option.  I have just been using the Climatic > Prepare > Extremes dialogue and this has the Missing checkbox, but no option to go to the summaries sub-dialogue.  But it might still be using the same "system" and so the missing options could also usefully be added here as well?

**Comment by dannyparsons** [comment]
Yes I think so. Let's list the dialogs where this needs to be added:

- [ ] Describe > One Variable > Summarise
- [ ] Describe > Two Variable > Summarise
- [ ] Climatic > Prepare > Extremes (possibly)

**Comment by dannyparsons** [comment]
This isn't urgent for the next release though.

**Comment by lilyclements** [comment]
@Ag-Derek can you assign this to someone for the "Describe > Two Variable > Summarise" option?

It is already implemented in "Describe > One Variable > Summarise", so hopefully follows on from that dialog and this looks to be pretty straight forward - especially since it is implemented elsewhere. 

The idea is to have the functionality of that "Omit Missing Values" checkbox, and then the "Options" and sdg too, like there is on the One Variable Summarise:

![Image](https://github.com/user-attachments/assets/f57a1061-f099-43dd-ac13-d731ac217ffe)

1. Checking the checkbox (Omit Missing Values) sets `na.rm = TRUE` in `data_book$summary_table`, otherwise if unchecked, `na.rm = FALSE`
2. If you click options, then additional bits come up, depending on what is clicked. This is still in the  `data_book$summary_table` function:
`na_type=c("'n'", "'n_non_miss'", "'prop'", "'con'")` if all four on the sdg are checked. If just the top one is checked then `na_type = c("'n'"), and so on.
Then we also get for the first nud: `na_max_prop = <NUD 1>`;
second nud: `na_max_n = <NUD 2>`
third nud: `na_min_n = <NUD 3>`
fourth nud: `na_consecutive_n = <NUD 4>`
